src/WorldStorage/FastNBT.cpp

```diff
@@ -638,15 +638,15 @@ void cFastNBTWriter::AddByteArray(const AString & a_Name, const char * a_Value,
 
 
 
-void cFastNBTWriter::AddIntArray(const AString & a_Name, const int * a_Value, size_t a_NumElements)
+void cFastNBTWriter::AddIntArray(const AString & a_Name, const Int32 * a_Value, size_t a_NumElements)
 {
 	TagCommon(a_Name, TAG_IntArray);
 	UInt32 len = htonl(static_cast<UInt32>(a_NumElements));
 	size_t cap = m_Result.capacity();
 	size_t size = m_Result.length();
 	if ((cap - size) < (4 + a_NumElements * 4))
 	{
-		m_Result.reserve(size + 4 + (a_NumElements * 4));
+		m_Result.reserve(size + 4 + (a_NumElements * 4));  // TODO: why _size_ + 4 ...?
```

Comment:
`size` for what already there, 4 bytes for the array length and `a_NumElements * 4` for the array elements.

Reply:
Thanks. I had the mistaken notion that reserve was relative to the current size instead of the new capacity...